src/agent/rustjail/src/cgroups/fs/mod.rs

```diff
@@ -1122,22 +1128,22 @@ impl Subsystem for Named {
     }
 }
 
-fn get_subsystem(name: &str) -> Result<Box<dyn Subsystem>> {
+fn get_subsystem(name: &str) -> Option<Box<dyn Subsystem>> {
     match name {
-        "cpuset" => Ok(Box::new(CpuSet())),
-        "cpu" => Ok(Box::new(Cpu())),
-        "devices" => Ok(Box::new(Devices())),
-        "memory" => Ok(Box::new(Memory())),
-        "cpuacct" => Ok(Box::new(CpuAcct())),
-        "pids" => Ok(Box::new(Pids())),
-        "blkio" => Ok(Box::new(Blkio())),
-        "hugetlb" => Ok(Box::new(HugeTLB())),
-        "net_cls" => Ok(Box::new(NetCls())),
-        "net_prio" => Ok(Box::new(NetPrio())),
-        "perf_event" => Ok(Box::new(PerfEvent())),
-        "freezer" => Ok(Box::new(Freezer())),
-        "name=systemd" => Ok(Box::new(Named())),
-        _ => Err(nix::Error::Sys(Errno::EINVAL).into()),
+        "cpuset" => Some(Box::new(CpuSet())),
+        "cpu" => Some(Box::new(Cpu())),
+        "devices" => Some(Box::new(Devices())),
+        "memory" => Some(Box::new(Memory())),
+        "cpuacct" => Some(Box::new(CpuAcct())),
+        "pids" => Some(Box::new(Pids())),
+        "blkio" => Some(Box::new(Blkio())),
+        "hugetlb" => Some(Box::new(HugeTLB())),
+        "net_cls" => Some(Box::new(NetCls())),
+        "net_prio" => Some(Box::new(NetPrio())),
+        "perf_event" => Some(Box::new(PerfEvent())),
+        "freezer" => Some(Box::new(Freezer())),
+        "name=systemd" => Some(Box::new(Named())),
+        _ => None,
```

Comment:
Thanks @lifupan ! Might be worth adding a log call so we can see the agent detected an unusual cgroup:

```rust
info!(sl!(), "Found unexpected cgroup"; "cgroup" => name);
```

Reply:
@jodh-intel  Done!

Reply:
Thanks @lifupan!